### PR TITLE
doc/dev/release-process.rst: modify for prerelease mechanism

### PR DESCRIPTION
**Pull request by dmick**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by dmick** [comment]
jenkins render docs

**Review by ljflores — APPROVED** [review]

File: doc/dev/release-process.rst [hunk 1]
```diff
@@ -205,14 +208,27 @@ See `the Ceph Tracker wiki page that explains how to write the release notes <ht
...
+
 5. Build Containers
 ===================
 
-Start the following two jobs:
+Prerelease containers (x86 only) are built by 
+https://2.jenkins.ceph.com/job/ceph-container-prerelease-build; run it 
+with appropriate parameters.  Test container images will appear on
+quay.ceph.io in the ceph/prerelease repo.  When satisfied with them,
+start the following two jobs:
```
Comment (yuriw) [inline doc/dev/release-process.rst:225]: I think we can start building the final containers after the `mv * ../` step

**Review by yuriw — DISMISSED** [review]

File: doc/dev/release-process.rst [hunk 2]
```diff
@@ -205,14 +208,27 @@ See `the Ceph Tracker wiki page that explains how to write the release notes <ht
...
+at https://download.ceph.com/prerelease.  Verify them from there.  
+When done and ready for release, mv the directories to the parent
+directory.  
+
+
 5. Build Containers
 ===================
 
-Start the following two jobs:
+Prerelease containers (x86 only) are built by
```
Comment (anthonyeleven) [inline doc/dev/release-process.rst:220]: x64?

Comment (dmick) [reply doc/dev/release-process.rst:220]: x86_64, but will change

File: doc/dev/release-process.rst (hunk 1 above)
Comment (dmick) [reply doc/dev/release-process.rst:225]: I suspect that you mean that we cannot start building the final containers *until* the mv step, and that's right and I'll add it

File: doc/dev/release-process.rst [hunk 3]
```diff
@@ -205,14 +208,27 @@ See `the Ceph Tracker wiki page that explains how to write the release notes <ht
 
       sync-push octopus
 
+This leaves the packages in a password-protected prerelease area
```
[the remark this answers is not in the document]
Comment (dmick) [reply doc/dev/release-process.rst:211]: ok

**Comment by dmick** [comment]
comments addressed, pls recheck

File: doc/dev/release-process.rst (hunk 1 above)
Comment (yuriw) [reply doc/dev/release-process.rst:226]: Yes thx

**Pull request merged**